test/release/examples/primers/learnChapelInYMinutes.chpl

```diff
@@ -850,19 +861,24 @@ class GenericClass {
   var classDomain: domain(1);
   var classArray: [classDomain] classType;
 
-// Explicit constructor.
-  proc GenericClass(type classType, elements : int) {
-    this.classDomain = {1..#elements};
+// Explicit initializer.
+  proc init(type classType, elements : int) {
+    this.classType = classType;
+    this.classDomain = {1..elements};
+    // all generic and const fields must be initialized in "phase 1" prior
+    // to a call to the superclass initializer.
+    super.init(); 
   }
 
-// Copy constructor.
-// Note: We still have to put the type as an argument, but we can
-// default to the type of the other object using the query (``?``) operator.
-// Further, we can take advantage of this to allow our copy constructor
-// to copy classes of different types and cast on the fly.
-  proc GenericClass(other : GenericClass(?otherType),
-                     type classType = otherType) {
+// Copy-style initializer.
+// Note: We include a type argument whose default is the type of the first
+// argument.  This lets our initializer copy classes of different
+// types and cast on the fly.
+  proc init(other : GenericClass(?),
+            type classType = other.classType) {
```

Comment:
I think this whole primer would really benefit from being rewritten by someone (and was thinking that a few people in a deep-dive doing it interactively might be a way to choke that medicine down), so think I'll leave it as-is for now and we can consider whether having this initializer in the primer is even high enough value to be worthwhile (I was trying to limit myself to obvious functionality and style improvements without getting sucked into the trap of reviewing the whole thing in this pass).